ui/src/app/auth/auth.component.ts

```diff
@@ -0,0 +1,23 @@
+import { Component }              from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Subscription }           from "rxjs/Rx";
+
+@Component({
+    selector: 'auth',
+    template: ''
+})
+
+export class AuthComponent {
+  private subscription: Subscription;
+  private token: string;
+
+  constructor(private router: Router, private route: ActivatedRoute) {
+    this.subscription = route.queryParams.subscribe(
+      (queryParam: any) => this.token = queryParam['token']
+    );
+
+    localStorage.setItem('id_token', this.token);
```

Comment:
By default the angular2-jwt package looks for a JWT named id_token. It can be changed but the package will need to be configured to look for the different name.